attestation-agent/protos/attestation-agent.proto

```diff
@@ -25,8 +25,28 @@ message ExtendRuntimeMeasurementRequest {
 
 message ExtendRuntimeMeasurementResponse {}
 
+message InitDataPlaintext {
+    bytes Content = 1;
+    string Algorithm = 2; 
+}
```

Comment:
So it's worth noting that this is InitData specific code. If we still haven't reached an agreement there, maybe we should leave support for the Plaintext to a later PR.

fwiw, I think this shows a clear benefit of the InitData approach. The policy does not include any information about how it is hashed and even if it did, we would have to do parsing of OPA to get it. We could add an additional field to pass the hash type from the shim (where the data is first hashed), to the kata agent, to the AA, to the KBS, but it seems much nicer to have it simply included in the InitData. cc @mkulke

Reply:
I don't mind having one endpoint for both even if only the digest is supported at first. I think we will need a plaintext option for the attestation flow to really add up, but this will require some agreement on the format.

Btw, @Xynnn007 this `InitDataPlaintext` message doesn't quite match up with your RFC spec which contains a version and digest field as well.

Reply:
> Btw, @Xynnn007 this InitDataPlaintext message doesn't quite match up with your RFC spec which contains a version and digest field as well.

Yes. I once hope that the format in RFC is handled inside kata-agent. That means
- version check
- data field into files
- use the digest field & hash alg to match the data field and call AA's `CheckInitData()` semantically /  DIrectly use data field and hash alg to call AA's `CheckInitDataPlaintext()` semantically, where the `InitDataPlaintext`

@mkulke I once thought that one API is enough, and to decrease the potential adaptation work when API changes, I used one API to receive different kinds of parameters as they both are "checking the initdata field binding relationship". 

But with your opinions I realize that the steps are probably relatively big. Let me add one simple parameter `initDataDigest` and left the plaintext extension in future together with the format issue.